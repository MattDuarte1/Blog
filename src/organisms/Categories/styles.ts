import styled, { css, DefaultTheme } from 'styled-components';

const BreakPoints = {
  ContainerSize: {
    medium: (theme: DefaultTheme) => css`
      @media ${theme.media.lteMedium} {
        padding: ${theme.spacings.medium};
      }
    `,
  },
  CategoryBodySize: {
    small: (theme: DefaultTheme) => css`
      @media ${theme.media.lteSmall} {
        div {
          width: 100%;
        }
      }
    `,
  },
};

export const Container = styled.section`
  ${({ theme }) => css`
    display: block;
    padding: ${theme.spacings.xxlarge};
    background-color: ${theme.colors.white};
    width: 100%;
    height: 100%;

    ${BreakPoints.ContainerSize['medium'](theme)}
  `}
`;

export const CategoryHeader = styled.div`
  display: flex;
  justify-content: center;
  margin-bottom: 2rem;
`;

export const CategoryBody = styled.section`
  ${({ theme }) => css`
    display: flex;
    flex-flow: row wrap;
    justify-content: center;
    min-width: 100%;
    height: 100%;
    gap: 37px;

    ${BreakPoints.CategoryBodySize['small'](theme)}
  `}
`;
