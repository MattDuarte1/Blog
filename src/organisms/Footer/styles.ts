import styled, { css } from 'styled-components';

export const Container = styled.footer`
  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    width: 100%;
    padding: 5rem;

    box-shadow: 1px 0 1rem 0 rgba(0, 0, 0, 0.2);

    display: flex;

    @media ${theme.media.lteMedium} {
      flex-direction: column;
      gap: 4rem;
      padding: 5rem;
    }
  `}
`;

export const Content = styled.aside`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    flex-direction: column;
    align-items: center;
    gap: ${theme.spacings.small};

    flex: 1;
  `}
`;

export const SocialMediaContent = styled.div`
  ${({ theme }) => css`
    display: flex;
    gap: 2rem;
    justify-content: center;
    width: 40%;
    a {
      color: ${theme.colors.black};
    }

    text-decoration: none;
    color: ${theme.colors.black};
    svg {
      cursor: pointer;
      font-size: 25px;
      transition: all 0.4s ease;

      &:hover {
        transform: scale(1.1);
      }
    }
  `}
`;
