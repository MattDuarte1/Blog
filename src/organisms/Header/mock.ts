export default {
  drawerIsOpen: false,
  menus: [
    { title: 'Inicio', newtab: false, href: '/#header' },
    { title: 'Sobre', newtab: false, href: '/#hero' },
    { title: 'Categoria', newtab: false, href: '/#categoria' },
    { title: 'Posts', newtab: false, href: '/#posts' },
  ],
};
