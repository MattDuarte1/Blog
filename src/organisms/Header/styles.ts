import styled, { css } from 'styled-components';

export const Container = styled.header`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: space-around;
    width: 100%;
    height: 86px;
    background-color: ${theme.colors.white};
    filter: drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.05));

    position: relative;
  `}
`;
