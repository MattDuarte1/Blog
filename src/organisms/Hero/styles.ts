import styled, { css, DefaultTheme, keyframes } from 'styled-components';

const BreakPoints = {
  Container: {
    medium: (theme: DefaultTheme) => css`
      @media ${theme.media.lteLarge} {
        padding: ${theme.spacings.huge} ${theme.spacings.medium};
      }
    `,
  },
  DescriptionSize: {
    small: (theme: DefaultTheme) => css`
      @media ${theme.media.lteSmall} {
        margin-top: 20px;
        font-size: ${theme.font.sizes.xsmall};
        line-height: normal;
      }
    `,
    medium: (theme: DefaultTheme) => css`
      @media ${theme.media.lteMedium} {
        margin-top: 40px;
        font-size: ${theme.font.sizes.small};
        line-height: normal;
      }
    `,
  },
  AsideRight: {
    medium: (theme: DefaultTheme) => css`
      @media ${theme.media.lteMedium} {
        img {
          width: 280px;
          height: 280px;
        }
      }
    `,
  },
};

const cursorEffect = (theme: DefaultTheme) => keyframes`
  0% {
    background-color: ${theme.colors.black};
  }
  100% {
    background-color: ${theme.colors.white};
  }
`;

export const Container = styled.section`
  ${({ theme }) => css`
    width: 100%;
    padding: ${theme.spacings.xhuge} ${theme.spacings.medium};
    min-height: 700px;
    display: flex;
    background-color: ${theme.colors.white};
    align-items: center;
    justify-content: center;
    flex-wrap: wrap;
    gap: 20px;

    ${BreakPoints.Container['medium'](theme)}
  `}
`;

export const AsideLeft = styled.aside`
  ${({ theme }) => css`
    max-height: 400px;
    max-width: 664px;
    padding: 25px
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    gap: 30px;

    span {
      color: transparent;
      width: 5px;
      height: 100%;
      width: 2px;
      animation: ${cursorEffect(theme)} 1s infinite;
    }
  `}
`;

export const AsideRight = styled(AsideLeft)`
  ${({ theme }) => css`
    justify-content: center;
    display: block;

    ${BreakPoints.AsideRight['medium'](theme)}
  `}
`;

export const Description = styled.p`
  ${({ theme }) => css`
    padding-left: 15px;
    margin-top: 25px;
    font-family: ${theme.font.family.default};
    font-style: normal;
    font-weight: 400;
    font-size: ${theme.font.sizes.medium};
    color: ${theme.colors.darkGrey};
    line-height: 25px;
    margin-left: 20px;
    position: relative;

    &::before {
      content: '';
      position: absolute;
      left: 0px;
      background-color: ${theme.colors.black};
      width: 5px;
      height: 100%;
    }

    ${BreakPoints.DescriptionSize['small'](theme)}
    ${BreakPoints.DescriptionSize['medium'](theme)}
  `}
`;
