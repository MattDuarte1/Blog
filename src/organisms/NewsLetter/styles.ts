import styled, { css, DefaultTheme } from 'styled-components';

const BreakPoints = {
  Content: (theme: DefaultTheme) => css`
    @media ${theme.media.lteMedium} {
      > h1 {
        font-size: ${theme.font.sizes.medium};
      }

      @media ${theme.media.lteSmall} {
        > h1 {
          font-size: ${theme.font.sizes.small};
        }
        > span {
          font-size: ${theme.font.sizes.xsmall};
        }
      }
    }
  `,
};

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    padding: 58px;
    gap: 10px;
    background-color: ${theme.name === 'inverted'
      ? theme.colors.white
      : theme.colors.lightGrey2};
    width: 100%;
  `}
`;

export const Content = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    padding: 0px;
    gap: 30px;

    max-width: 696px;

    ${BreakPoints.Content(theme)}
  `}
`;

export const Form = styled.form`
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  align-items: center;
  padding: 0px;
  gap: 20px;
  width: 100%;
`;

export const InputForm = styled.input`
  display: flex;
  flex-direction: row;
  align-items: center;
  gap: 10px;
  padding: 21px 21px 21px 25px;

  width: 300px;
  height: 67px;

  /* White */

  background: #fffffe;
  border-radius: 8px;
  border: none;

  &:focus {
    box-shadow: 0 0 0 0;
    outline: 0;
  }
`;
