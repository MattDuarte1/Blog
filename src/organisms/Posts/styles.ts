import styled, { css, DefaultTheme } from 'styled-components';

const MediaBreakPoints = {
  small: (theme: DefaultTheme) => css`
    @media ${theme.media.lteSmall} {
      padding: 3.5rem;
    }
  `,
};

export const Container = styled.div`
  ${({ theme }) => css`
    padding: 4.4rem;
    background-color: ${theme.colors.white};
    ${MediaBreakPoints.small(theme)}
  `}
`;

export const Content = styled.div`
  ${({ theme }) => css`
    margin-top: 1.4rem;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(26rem, 2fr));
    gap: ${theme.spacings.large};

    @media (max-width: 1065px) {
      grid-template-columns: repeat(auto-fill, minmax(27rem, 1fr));
    }

    @media (max-width: 705px) {
      justify-content: center;
      grid-template-columns: 0fr;
    }
  `}
`;
