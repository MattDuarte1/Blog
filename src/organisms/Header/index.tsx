import { forwardRef, useState } from 'react';
import Logo from '@/atoms/Logo';
import NavBar from '@/molecules/NavBar';
import { IMenu } from '@/interfaces/Menu';
import * as Styled from './styles';
import ToggleDrawer from '@/atoms/ToggleDrawer';
import Link from 'next/link';
import { useBlogContext } from '@/contexts/Theme/hook';

interface StateMenuInterface {
  drawerIsOpen: boolean;
  menus: IMenu[];
}

const Header = forwardRef<HTMLElement>((props, ref) => {
  const [data, _] = useState<StateMenuInterface>({
    drawerIsOpen: false,
    menus: [
      { title: 'Inicio', newtab: false, href: '/#header' },
      { title: 'Sobre', newtab: false, href: '/#hero' },
      { title: 'Categoria', newtab: false, href: '/#categoria' },
      { title: 'Posts', newtab: false, href: '/#posts' },
    ],
  });
  const {
    state: { drawerIsOpen },
    activeDrawer,
  } = useBlogContext();

  return (
    <Styled.Container id="header" ref={ref} {...props}>
      <Link href={'/'}>
        <Logo />
      </Link>
      <NavBar menus={data.menus} />
      <ToggleDrawer
        activeDrawer={() => activeDrawer()}
        drawerIsOpen={drawerIsOpen}
      />
    </Styled.Container>
  );
});

Header.displayName = 'Header Component';

export default Header;
