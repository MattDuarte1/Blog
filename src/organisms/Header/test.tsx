import Header from '.';
import { renderTheme } from '@/styles/render-theme';
import React from 'react';
import { fireEvent, screen } from '@testing-library/react';
import { useSession } from 'next-auth/react';

jest.mock('next-auth/react');
const mockUserSession = useSession as jest.Mock;
mockUserSession.mockReturnValue({
  status: 'unauthenticated',
  data: null,
});

describe('<Header />', () => {
  it('Should render default header component with menus when toggle mobile is not visible', () => {
    renderTheme(<Header />);
    const logo = screen.getByText(/blogzin/i);
    const menus = screen.getAllByRole('link');

    expect(logo).toBeInTheDocument();
    expect(screen.getByRole('link', { name: /blogzin/i })).toHaveAttribute(
      'href',
      '/',
    );
    expect(screen.getByLabelText('menu-toggle')).toHaveStyle({
      display: 'none',
    });
    expect(menus).toHaveLength(5);
  });

  it('Should call ActiveDrawer function when togglerDrawer is clicked', () => {
    renderTheme(<Header />);
    const toggleDrawer = screen.getByLabelText('menu-toggle');

    expect(toggleDrawer).toBeInTheDocument();

    fireEvent.click(toggleDrawer);
  });

  it('Should match snapshot component', () => {
    const { container } = renderTheme(<Header />);

    expect(container).toMatchSnapshot();
  });
});
