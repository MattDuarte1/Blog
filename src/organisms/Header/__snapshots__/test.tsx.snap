// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Should match snapshot component 1`] = `
.c1 {
  font-family: 'Plus Jakarta Sans','sans-serif';
  font-style: normal;
  font-weight: 600;
  font-size: 3.576rem;
  line-height: 45px;
  position: relative;
}

.c1::first-letter {
  color: #6246EA;
}

.c1::after {
  content: '.blog';
  font-family: 'Plus Jakarta Sans','sans-serif';
  font-style: normal;
  font-weight: 600;
  font-size: 2.156rem;
  color: #6246EA;
}

.c3 {
  font-family: 'Plus Jakarta Sans','sans-serif';
  font-style: normal;
  font-weight: 600;
  font-size: 1.617rem;
  line-height: 25px;
  text-transform: capitalize;
  color: #2B2C34;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 10px;
  max-width: 100px;
  height: 25px;
}

.c4 {
  background: #6246EA;
  border: none;
  border-radius: 25px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.9rem;
  padding: 1rem;
  cursor: pointer;
}

.c5 {
  font-family: 'Plus Jakarta Sans','sans-serif';
  font-style: normal;
  font-weight: 700;
  line-height: 1.2;
  text-transform: capitalize;
  font-size: 1.617rem;
  text-transform: none;
  color: #2B2C34;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 44px;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0px;
  gap: 50px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  width: 100%;
  height: 86px;
  background-color: #FFFFFE;
  -webkit-filter: drop-shadow(0px 4px 4px rgba(0,0,0,0.05));
  filter: drop-shadow(0px 4px 4px rgba(0,0,0,0.05));
  position: relative;
}

.c6 {
  background-color: #fff;
  box-shadow: 0 10px 20px rgba(0,0,0,0.08);
  border-radius: 10px;
  border: none;
  overflow: hidden;
  display: none;
  cursor: pointer;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  height: 70px;
  width: 70px;
}

.c6 p {
  position: absolute;
  width: 40px;
  height: 4px;
  background-color: #6246EA;
  border-radius: 4px;
  -webkit-transition: 0.5s;
  transition: 0.5s;
}

.c6 p:nth-child(1) {
  -webkit-transform: translateY(-15px);
  -ms-transform: translateY(-15px);
  transform: translateY(-15px);
  width: 25px;
  left: 15px;
}

.c6 p:nth-child(2) {
  -webkit-transform: translateY(15px);
  -ms-transform: translateY(15px);
  transform: translateY(15px);
  width: 15px;
  left: 15px;
}

@media (max-width:999px) {
  .c2 {
    display: none;
  }
}

@media (max-width:999px) {
  .c6 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<div>
  <header
    class="c0"
    id="header"
  >
    <a
      href="/"
    >
      <div
        class="c1"
      >
        BlogZin
      </div>
    </a>
    <nav
      aria-label="navbar"
      class="c2"
    >
      <a
        class="c3"
        href="/#header"
        target="_self"
      >
        Inicio
      </a>
      <a
        class="c3"
        href="/#hero"
        target="_self"
      >
        Sobre
      </a>
      <a
        class="c3"
        href="/#categoria"
        target="_self"
      >
        Categoria
      </a>
      <a
        class="c3"
        href="/#posts"
        target="_self"
      >
        Posts
      </a>
      <button
        class="c4"
      >
        <svg
          fill="currentColor"
          height="30"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 480 512"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M186.1 328.7c0 20.9-10.9 55.1-36.7 55.1s-36.7-34.2-36.7-55.1 10.9-55.1 36.7-55.1 36.7 34.2 36.7 55.1zM480 278.2c0 31.9-3.2 65.7-17.5 95-37.9 76.6-142.1 74.8-216.7 74.8-75.8 0-186.2 2.7-225.6-74.8-14.6-29-20.2-63.1-20.2-95 0-41.9 13.9-81.5 41.5-113.6-5.2-15.8-7.7-32.4-7.7-48.8 0-21.5 4.9-32.3 14.6-51.8 45.3 0 74.3 9 108.8 36 29-6.9 58.8-10 88.7-10 27 0 54.2 2.9 80.4 9.2 34-26.7 63-35.2 107.8-35.2 9.8 19.5 14.6 30.3 14.6 51.8 0 16.4-2.6 32.7-7.7 48.2 27.5 32.4 39 72.3 39 114.2zm-64.3 50.5c0-43.9-26.7-82.6-73.5-82.6-18.9 0-37 3.4-56 6-14.9 2.3-29.8 3.2-45.1 3.2-15.2 0-30.1-.9-45.1-3.2-18.7-2.6-37-6-56-6-46.8 0-73.5 38.7-73.5 82.6 0 87.8 80.4 101.3 150.4 101.3h48.2c70.3 0 150.6-13.4 150.6-101.3zm-82.6-55.1c-25.8 0-36.7 34.2-36.7 55.1s10.9 55.1 36.7 55.1 36.7-34.2 36.7-55.1-10.9-55.1-36.7-55.1z"
          />
        </svg>
        <h1
          class="c5"
        >
          Conectar com Github
        </h1>
      </button>
    </nav>
    <button
      aria-label="menu-toggle"
      class="c6"
      data-active="false"
    >
      <p />
      <p />
      <p />
    </button>
  </header>
</div>
`;
